fili-core/src/test/groovy/com/yahoo/bard/webservice/async/AsyncResultsReadySpec.groovy

```diff
@@ -0,0 +1,99 @@
+
+/**
+ * Verifies that when the results of an asynchronous request are ready, that the job status is updated to "success" and
+ * the results are accessible through both the {@code results} and {@code syncResults} links.
+ */
+class AsyncResultsReadySpec extends AsyncFunctionalSpec {
+    static final String QUERY =
+            "http://localhost:9998/data/shapes/day?dateTime=2016-08-30%2F2016-08-31&metrics=height&asyncAfter=0"
+
+    @Override
+    Map<String, Closure<String>> getResultsToTargetFunctions() {
+        [
+                data: { "data/shapes/day" },
+                //By querying the syncResults link first, we wait until the results are ready, thanks to the
+                //BroadcastChannel. Thus, we wait until the results are ready before verifying that the job status
+                //has been updated correctly.
+                syncResults: { AsyncTestUtils.extractTargetFromField(it.data.readEntity(String), "syncResults") },
+                results: { AsyncTestUtils.extractTargetFromField(it.data.readEntity(String), "results") },
+                jobs: { AsyncTestUtils.buildTicketLookup(it.data.readEntity(String)) }
+        ]
+    }
+
+    @Override
+    Map<String, Closure<Void>> getResultAssertions() {
+        [
+                data: {
+                    assert it.status == 202
+                    AsyncTestUtils.validateJobPayload(it.readEntity(String), QUERY, DefaultJobStatus.PENDING.name)
+                },
+                syncResults: {
+                    assert it.status == 200
+                    assert GroovyTestUtils.compareJson(it.readEntity(String), getExpectedApiResponse())
+                },
+                results: {
+                    assert it.status == 200
+                    assert GroovyTestUtils.compareJson(it.readEntity(String), getExpectedApiResponse())
+                },
+                jobs: { response ->
+                    assert response.status == 200
+                    AsyncTestUtils.validateJobPayload(
+                            response.readEntity(String),
+                            QUERY,
+                            DefaultJobStatus.SUCCESS.name
+                    )
+                }
+        ]
+    }
+
+    @Override
+    Map<String, Closure<Map<String, List<String>>>> getQueryParameters() {
+        [
+                data: {[
+                        metrics: ["height"],
+                        asyncAfter: ["0"],
+                        dateTime: ["2016-08-30/2016-08-31"]
+                ]},
+                syncResults: {
+                    new URI(AsyncTestUtils.getJobFieldValue(it.data.readEntity(String), "syncResults"))
+                            .query.split("&")
+                            .collectEntries { parameter ->
+                                    def (String key, String value) = parameter.tokenize("=")
+                                    return [(key): value.tokenize(",")]
+                            }
```

Comment:
Does it make sense to extract this bit as a helper method as well? I think I saw it at least 2 places in this PR and it's not the most easy-to-understand code under the sun.